avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/AvroSchemaHelper.java

```diff
@@ -24,6 +24,8 @@
 
 public abstract class AvroSchemaHelper
 {
+    private static final Map<String, String> SCHEMA_NAME_CACHE = new HashMap<>();
```

Comment:
I have some concerns about this.

Since it is `static` it is per-ClassLoader, and ideally nothing in Jackson would use that. If at all possible this should be tied to something else, probably `AvroFactory` (or `AvroMapper` if practical).

Second, all caches should be bound (not with unlimited size). This is easy to solve by just using

    com.fasterxml.jackson.databind.util.LRUMap

with whatever size settings (default, max size) makes sense.

Reply:
Also another significant problem is that access is not synchronized: since this will get called from multiple threads, and it is not read-only, access must be made thread-safe. `LRUMap` uses `ConcurrentHashMap` which solves that issue.

Reply:
Question from a Java non-expert: will the call `Class.forName(nestedClassName)` (called in the method that resolves the name) use the default class loader?  If so, no matter how we cache the names, it will still be broken.

Reply:
Yes, that would use the default class loader of... well. It's probably the context class loader (alternative being class loader that loaded given class; usually these are the same but not always).